srml/staking/src/lib.rs

```diff
@@ -514,50 +711,19 @@ impl<T: Trait> Module<T> {
 			});
 		}
 
-		let event = if new_slash_count > grace {
-			let slash = {
-				let base_slash = Self::current_offline_slash();
-				let instances = slash_count - grace;
-
-				let mut total_slash = BalanceOf::<T>::default();
-				for i in instances..(instances + count as u32) {
-					if let Some(total) = base_slash.checked_shl(i)
-							.and_then(|slash| total_slash.checked_add(&slash)) {
-						total_slash = total;
-					} else {
-						// reset slash count only up to the current
-						// instance. the total slash overflows the unit for
-						// balance in the system therefore we can slash all
-						// the slashable balance for the account
-						<SlashCount<T>>::insert(v.clone(), slash_count + i);
-						total_slash = Self::slashable_balance(&v);
-						break;
-					}
-				}
-
-				total_slash
-			};
+		let prefs = Self::validators(&v);
+		let unstake_threshold = prefs.unstake_threshold.min(MAX_UNSTAKE_THRESHOLD);
+		let max_slashes = grace + unstake_threshold;
 
+		let event = if new_slash_count > max_slashes {
+			// They're bailing.
+			let slash = Self::current_offline_slash()
+				.checked_shl(prefs.unstake_threshold)
```

Comment:
`prefs.unstake_threshold` is a constant, and does not make sense (to me) to be used as the exponential weighting factor multiplied by `current_offline_slash`. Instead we should use the `new_slash` value, which represents the total number of `slashes` for an account at the time of executing this logic

Reply:
I could see this making sense if the validator is removed at that point, thus only a single slashing will occur, and should be at the `unstake_threshold` amount.

Reply:
I don't think the code at least implies that the slashed account will not be a validator any more. Based on how much nomination it has, it can still be chosen for the next era/round. Looking at the function `on_offline_validator()` -> `slash_validator()` trace I only see the account being removed from `<Validators<T>>` storage, which gets re-populated at each era.